src/openforms/registrations/utils.py

```diff
@@ -30,3 +33,34 @@ def execute_unless_result_exists(
     assign(submission.registration_result, spec, result, missing=dict)
     submission.save(update_fields=["registration_result"])
     return callback_result
+
+
+def collect_registrations_failures(since: datetime) -> list[dict[str, str] | None]:
```

Comment:
this would also benefit from using a dataclass for the processed log data instead of `dict[str, str]` (which is also a lie because some values are `int` and `datetime` rather than `str`)